astropy/config/paths.py

```diff
@@ -144,7 +118,11 @@ def get_cache_dir(rootname: str = "astropy") -> str:
 
 class _SetTempPath:
     _temp_path: Path | None = None
-    _default_path_getter: Callable[[str], str]
+
+    # This base class serves as a deduplication layer for its only two intended
+    # children (set_temp_cache and set_temp_config)
+    _directory_type: Literal["cache", "config"]
+    _xdg_directory_env_var: Literal["XDG_CACHE_HOME", "XDG_CONFIG_HOME"]
```

Comment:
Given the plan to follow up with https://github.com/astropy/astropy/pull/17567 for a different env var name, maybe no need to hardcode `xdg` into the new class attribute now?

Reply:
done